src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -69,7 +74,212 @@ trait SubmissionComponent {
     def value = foreignKey("FK_SUB_VALIDATION_VAL", valueId, attributeQuery)(_.id.?)
   }
 
-  protected val submissionQuery = TableQuery[SubmissionTable]
   protected val submissionValidationQuery = TableQuery[SubmissionValidationTable]
 
+  object submissionQuery extends TableQuery(new SubmissionTable(_)) {
+
+    type SubmissionQueryType = Query[SubmissionTable, SubmissionRecord, Seq]
+
+    /*
+      the core methods
+     */
+
+    /* gets a submission */
+    def get(workspaceContext: SlickWorkspaceContext, submissionId: String): ReadAction[Option[Submission]] = {
+      loadSubmission(UUID.fromString(submissionId))
+    }
+
+    /* lists all submissions in a workspace */
+    def list(workspaceContext: SlickWorkspaceContext): ReadAction[Seq[Submission]] = {
+      findByWorkspaceId(workspaceContext.workspaceId).result.flatMap(recs => DBIO.sequence(recs.map{ rec =>
+        loadSubmission(rec.id) map(sub => sub.get)
```

Comment:
I don't like this dot-get but I am struggling to find a better solution at the moment